SmartDeviceLink/SDLPermissionManager.m

```diff
@@ -287,6 +313,39 @@ - (void)sdl_hmiLevelDidChange:(SDLRPCNotificationNotification *)notification {
 
 #pragma mark Helper Methods
 
+/**
+ *  HAX: Remove this when statusOfRPCs: is no longer supported. Converts a dictionary from <SDLRPCFunctionName, SDLRPCPermissionStatus*> to <SDLPermissionRPCName, NSNumber *>.
+ *
+ *  @param permissionStatusDictionary The dictionary containing <SDLRPCFunctionName, SDLRPCPermissionStatus*> to convert.
+ *
+ *  @return A  <SDLPermissionRPCName, NSNumber *> dictionary.
+ */
+- (NSDictionary<SDLPermissionRPCName, NSNumber *> *)sdl_convertPermissionsDictionary:(NSDictionary<SDLRPCFunctionName, SDLRPCPermissionStatus*> *)permissionStatusDictionary {
+    NSMutableDictionary *rpcNameDictionary = [[NSMutableDictionary alloc] init];
+    [permissionStatusDictionary enumerateKeysAndObjectsUsingBlock:^(SDLRPCFunctionName _Nonnull key, SDLRPCPermissionStatus * _Nonnull obj, BOOL * _Nonnull stop) {
+        [rpcNameDictionary setObject:@(obj.rpcAllowed) forKey:key];
+    }];
+
+    return rpcNameDictionary;
+}
+
+/**
+ *  Converts an array of RPC name strings to permission elements.
+ *
+ *  @param rpcNames The RPC names to convert.
+ *
+ *  @return An array of permission elements.
+ */
+- (NSArray<SDLPermissionElement *> *)sdl_createPermissionElementFromRPCNames:(NSArray<NSString *> *)rpcNames {
+    NSMutableArray *permissionElements = [[NSMutableArray alloc] init];
+    for (NSString *rpcName in rpcNames) {
+        SDLPermissionElement *permissionElement = [[SDLPermissionElement alloc] initWithRPCName:rpcName parameterPermissions:nil];
+        [permissionElements addObject:permissionElement];
+    }
+
+    return permissionElements;
```

Comment:
```suggestion
    return [permissionElements copy];
```

If you don't copy this you sneakily have a mutable array and bad stuff could occur.